units/enemy.gd

```diff
@@ -2,7 +2,12 @@ extends Node2D
 
 
 const SPEED = 150
+const DESPAWN_BOUND = -50
 
 
 func _physics_process(delta):
 	position -= transform.x * SPEED * delta
+	
+	# despawn enemies that have moved beyond the left of the screen
+	if global_position.x < DESPAWN_BOUND:
```

Comment:
https://docs.godotengine.org/en/stable/classes/class_visibleonscreennotifier2d.html

we should avoid re-inventing the wheel :grin:

Reply:
If nothing else this game dev is super helpful to learn how to actually use godot :nods: